src/DynamoUtilities/EnumerableExtensions.cs

```diff
@@ -467,5 +468,30 @@ where attribute.Name.Equals("value")
 
             return string.Empty;
         }
+
+        public static bool TryGetValueAs<Key, Value, ValueAs>(this IDictionary<Key, Value> dictionary, Key key, out ValueAs valueAs) where ValueAs : Value
+        {
+            if (dictionary.TryGetValue(key, out Value value))
+            {
+                valueAs = (ValueAs)value;
```

Comment:
Is it possible that this cast fails in some cases?

Reply:
It is possible...I'll adjust the code to handle that